include/kframework/infinite-gas.md

```diff
@@ -5,134 +5,49 @@ Here we use the construct `#gas` to represent positive infinity, while tracking
 This allows (i) computing final gas used, and (ii) never stopping because of out-of-gas.
 Note that the argument to `#gas(_)` is just metadata tracking the current gas usage, and is not meant to be compared to other values.
 As such, any `#gas(G)` and `#gas(G')` are the _same_ positive infinite, regardless of the values `G` and `G'`.
-In particular, this means that `#gas(_) <Int #gas(_) => false`, and `#gas(_) <=Int #gas(_) => true`, regardless of the values contained in the `#gas(_)`.
+In particular, this means that `#gas(_) <Gas #gas(_) => false`, and `#gas(_) <=Gas #gas(_) => true`, regardless of the values contained in the `#gas(_)`.
 
 ```k
 requires "evm.md"
 
 module INFINITE-GAS
-    imports INFINITE-GAS-HASKELL
-endmodule
-
-module INFINITE-GAS-HASKELL [kore]
-    imports INFINITE-GAS-COMMON
-
-    rule #gas(_) <=Int _C => false [concrete(_C), simplification]
-    rule _C <=Int #gas(_) => true  [concrete(_C), simplification]
-
-    rule { #gas(_) #Equals #gas(_) } => #Top [simplification]
-endmodule
-
-module INFINITE-GAS-COMMON
+    imports BASIC-K
+    imports INT
+    imports BOOL
+    imports GAS
     imports EVM
 
- // Generic rules
-
-    rule (notBool (A andBool B)) andBool A => (notBool B) andBool A                      [simplification]
-    rule #if B #then C +Int C1 #else C +Int C2 #fi => C +Int #if B #then C1 #else C2 #fi [simplification]
-
- // Symbolic Gas
-
-    syntax Int ::= #gas ( Int ) [function, total, no-evaluators, klabel(infGas), symbol, smtlib(infGas)]
- // ----------------------------------------------------------------------------------------------------
-    rule #gas(G) +Int G' => #gas(G +Int G') requires 0 <=Int G' orBool 0 -Int G' <Int #gas(G)  [simplification]
-    rule G +Int #gas(G') => #gas(G +Int G') requires 0 <=Int G  orBool 0 -Int G  <Int #gas(G') [simplification]
+    syntax Gas ::= #gas(Int)
```

Comment:
Are there any modifiers that should be used for this? I see that the last version used `[symbol]` and `[smtlib(infGas)]` etc

Reply:
We should probably give it `smtlib(infGas), symbol, klabel(infGas)`. The `symbol, klabel(infGas)` is so that [here](https://github.com/runtimeverification/evm-semantics/blob/3a4aaf884ff2df3455bd8f78b4a579087ea74e5f/kevm-pyk/src/kevm_pyk/kevm.py#L255) we get the correct AST node where building them programatically.

The `smtlib` is so that Z3 knows that this behaves like a function. It will get slightly better reasoning out of it that way.